services/core/java/com/android/server/mperspective/PerspectiveService.java

```diff
@@ -332,6 +351,10 @@ private void startDesktopPerspectiveInternal() {
                         updateDesktopStateLocked(Perspective.STATE_STOPPED);
                     }
                 }
+                // After starting desktop, we should make sure the desktop
+                // input is disabled. If in some occasion, we want to use
+                // input in desktop, please enable it manually.
+                updateDesktopInteractiveStateInternal(false);
```

Comment:
I think we need to check whether a public presentation display is available and then set the interactive state based on that. If the user stops the desktop manually and then starts it again manually while a public presentation display is connected, this will disable the input for the desktop even though it should be enabled since a display is available. The user will need to disconnect and re-connect the display to fix this.

Reply:
I remove logic to `updateDesktopStateLocked`, and combine public presentation display connecting state and desktop running state to set desktop interactive state.